algebra2/gaussian_elimination/gaussian_elimination.hpp

```diff
@@ -290,273 +347,281 @@ namespace algorithms::gaussian_elimination {
-             axis a,
              reducted_form reducted = reducted_form::echelon,
-             operation allowed_operations = operation::swap | operation::add | operation::multiply){
-        auto [m_reduced_vector, m_reduced] = gaussian_elimiantion_alg(m, axis::rows, reducted).matrix_range;
-        
+             operation allowed_operations = operation::swap | operation::add |
+                                            operation::multiply)
+        -> std::pair<std::vector<fraction<T>>,
+                     ranges::matrix_view<fraction<T>, LP>> {
+        auto [m_reduced_vector, m_reduced] =
+            gaussian_elimiantion_alg(m, reducted, allowed_operations)
+                .matrix_range;
+
         return std::pair{std::move(m_reduced_vector), m_reduced};
     }
 
+
     /*
         description:
-            runs gaussian elimination on axis a for matrix m until the reducted form reducted is obtained
-            demonstrates on screen intermediate steps of the elimination
+            runs gaussian elimination on axis a for matrix m until the reducted
+       form reducted is obtained demonstrates on screen intermediate steps of
+       the elimination
     */
-    template<typename T, typename LP>
+    template <typename T, typename LP>
     auto show_steps(ranges::matrix_view<T, LP> m,
-                    axis a,
                     reducted_form reducted = reducted_form::echelon,
-                    operation allowed_operations = operation::swap | operation::add | operation::multiply) -> void{
-        std::vector<std::string> steps = gaussian_elimiantion_alg(m, axis::rows, reducted).reduction_steps;
+                    operation allowed_operations = operation::swap |
+                                                   operation::add |
+                                                   operation::multiply)
+        -> void {
+        std::vector<std::string> steps =
+            gaussian_elimiantion_alg(m, reducted, allowed_operations)
+                .reduction_steps;
         std::print("{}\n", steps);
     }
 
+
     /*
         description:
             calculates the determinant of matrix m
     */
     template <typename T, typename LP>
-    auto determinant(ranges::matrix_view<T, LP> m) -> std::expected<fraction<int>, error>{
-        int rows = m.extent(0),
-            cols = m.extent(1);
-
-        if(rows != cols){
-            return std::unexpected(error::not_square);
-        }else{
-            return gaussian_elimiantion_alg(m, axis::rows, reducted_form::diagonal).determinant_m;
-        }
+    auto determinant(ranges::matrix_view<T, LP> m)
+        -> std::expected<fraction<int>, error> {
+        const std::size_t rows = m.number_of_rows();
+        const std::size_t cols = m.number_of_columns();
+
+        if (rows != cols) { return std::unexpected(error::not_square); }
+        return gaussian_elimiantion_alg(m, reducted_form::diagonal)
+            .determinant_m;
```

Comment:
this should be in namespace utils::matrix  
similarly with other matrix methods